redbot/cogs/mod/kickban.py

```diff
@@ -658,16 +671,38 @@ async def tempban(
 
         with contextlib.suppress(discord.HTTPException):
             # We don't want blocked DMs preventing us from banning
-            msg = _("You have been temporarily banned from {server_name} until {date}.").format(
-                server_name=guild.name, date=discord.utils.format_dt(unban_time)
+
+            extra_embed = await self.config.guild(guild).ban_show_extra()
+
+            em = discord.Embed(
+                title=bold(
+                    _("You have been temporarily banned from {guild} until {date}.").format(
+                        guild=guild, date=discord.utils.format_dt(unban_time)
+                    )
+                ),
+                color=await self.bot.get_embed_color(member),
+            )
+            em.add_field(
+                name=_("**Reason**"),
+                value=reason if reason is not None else _("No reason was given."),
+                inline=False,
             )
-            if guild_data["dm_on_kickban"] and reason:
-                msg += _("\n\n**Reason:** {reason}").format(reason=reason)
             if invite:
-                msg += _("\n\nHere is an invite for when your ban expires: {invite_link}").format(
-                    invite_link=invite
+                em.add_field(
+                    name=bold(_("Here is an invite for when your ban expires")),
+                    value=invite,
+                    inline=False,
+                )
+            if extra_embed:
+                extra_embed_title = await self.config.guild(guild).ban_extra_embed_title()
+                extra_embed_contents = await self.config.guild(guild).ban_extra_embed_contents()
+
+                em.add_field(
+                    name=bold(extra_embed_title),
+                    value=extra_embed_contents,
+                    inline=False,
                 )
```

Comment:
> From what I can tell, `bold` just adds the 4 stars to bold a message so that leaves us with 252 characters. From some testing, `len` does not count escape characters and counts what they become (so \n would be 1)

In the other ban case, you passed `False` as the second parameter to `bold`. That is probably why your testing showed that escape characters were not counted, since they were never included at all. In this case (tempbans), formatting is escaped, which results in cases where the ban DM is not sent due to the suppressed `discord.HTTPException` triggering when the resulting title exceeds the character count. You need to also disable escaping logic here to avoid issues.

```suggestion
                em.add_field(
                    name=bold(extra_embed_title, escape_formatting=False),
                    value=extra_embed_contents,
                    inline=False,
                )
```